### PR TITLE
feat(Pagination): add pagination component

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Adds new `Pagination` component

### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=8639%3A7128) 
- [x] Related to JIRA ticket: gm-355
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.38-alpha.1b5cfe.0<br/>@codecademy/gamut-labs@28.6.1-alpha.1b5cfe.0<br/>@codecademy/gamut@42.2.2-alpha.1b5cfe.0<br/>@codecademy/monaco@0.5.28-alpha.1b5cfe.0<br/>@codecademy/styleguide@52.4.1-alpha.1b5cfe.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d38a4405e7dc1a0657e5fe--gamut-preview.netlify.app>https://61d38a4405e7dc1a0657e5fe--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d38a4405e7dc1a0657e5fe>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.38-alpha.32b789.0<br/>@codecademy/gamut-labs@28.6.1-alpha.32b789.0<br/>@codecademy/gamut@42.2.2-alpha.32b789.0<br/>@codecademy/monaco@0.5.28-alpha.32b789.0<br/>@codecademy/styleguide@52.4.1-alpha.32b789.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d472a2529d713acb6e2721--gamut-preview.netlify.app>https://61d472a2529d713acb6e2721--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d472a2529d713acb6e2721>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.38-alpha.065e6a.0<br/>@codecademy/gamut-labs@28.6.1-alpha.065e6a.0<br/>@codecademy/gamut@42.2.2-alpha.065e6a.0<br/>@codecademy/monaco@0.5.28-alpha.065e6a.0<br/>@codecademy/styleguide@52.4.1-alpha.065e6a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4746b0327743e79ae88d5--gamut-preview.netlify.app>https://61d4746b0327743e79ae88d5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4746b0327743e79ae88d5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.38-alpha.524bc2.0<br/>@codecademy/gamut-labs@28.6.1-alpha.524bc2.0<br/>@codecademy/gamut@42.2.2-alpha.524bc2.0<br/>@codecademy/monaco@0.5.28-alpha.524bc2.0<br/>@codecademy/styleguide@52.4.1-alpha.524bc2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4993ff348b400995e2a2d--gamut-preview.netlify.app>https://61d4993ff348b400995e2a2d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4993ff348b400995e2a2d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.ebdee8.0<br/>@codecademy/gamut-kit@0.6.38-alpha.ebdee8.0<br/>@codecademy/gamut-labs@28.6.1-alpha.ebdee8.0<br/>@codecademy/gamut@42.3.0-alpha.ebdee8.0<br/>@codecademy/monaco@0.5.28-alpha.ebdee8.0<br/>@codecademy/styleguide@52.4.1-alpha.ebdee8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4abe9141e31185a3a1243--gamut-preview.netlify.app>https://61d4abe9141e31185a3a1243--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4abe9141e31185a3a1243>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.46f590.0<br/>@codecademy/gamut-kit@0.6.38-alpha.46f590.0<br/>@codecademy/gamut-labs@28.6.1-alpha.46f590.0<br/>@codecademy/gamut@42.3.0-alpha.46f590.0<br/>@codecademy/monaco@0.5.28-alpha.46f590.0<br/>@codecademy/styleguide@52.4.1-alpha.46f590.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.18eb14.0<br/>@codecademy/gamut-kit@0.6.38-alpha.18eb14.0<br/>@codecademy/gamut-labs@28.6.1-alpha.18eb14.0<br/>@codecademy/gamut@42.3.0-alpha.18eb14.0<br/>@codecademy/monaco@0.5.28-alpha.18eb14.0<br/>@codecademy/styleguide@52.4.1-alpha.18eb14.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d5f9d46392d8a636984e7e--gamut-preview.netlify.app>https://61d5f9d46392d8a636984e7e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d5f9d46392d8a636984e7e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.8d4182.0<br/>@codecademy/gamut-kit@0.6.38-alpha.8d4182.0<br/>@codecademy/gamut-labs@28.6.1-alpha.8d4182.0<br/>@codecademy/gamut@42.3.0-alpha.8d4182.0<br/>@codecademy/monaco@0.5.28-alpha.8d4182.0<br/>@codecademy/styleguide@52.4.1-alpha.8d4182.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d6513b95fdea1209711d35--gamut-preview.netlify.app>https://61d6513b95fdea1209711d35--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d6513b95fdea1209711d35>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.f84685.0<br/>@codecademy/gamut-kit@0.6.38-alpha.f84685.0<br/>@codecademy/gamut-labs@28.6.1-alpha.f84685.0<br/>@codecademy/gamut@42.3.0-alpha.f84685.0<br/>@codecademy/monaco@0.5.28-alpha.f84685.0<br/>@codecademy/styleguide@52.4.1-alpha.f84685.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d6550059add3105c2e4d2f--gamut-preview.netlify.app>https://61d6550059add3105c2e4d2f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d6550059add3105c2e4d2f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.33f70c.0<br/>@codecademy/gamut-kit@0.6.38-alpha.33f70c.0<br/>@codecademy/gamut-labs@28.6.1-alpha.33f70c.0<br/>@codecademy/gamut@42.3.0-alpha.33f70c.0<br/>@codecademy/monaco@0.5.28-alpha.33f70c.0<br/>@codecademy/styleguide@52.5.0-alpha.33f70c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d657e095fdea1860711cbc--gamut-preview.netlify.app>https://61d657e095fdea1860711cbc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d657e095fdea1860711cbc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.c2129d.0<br/>@codecademy/gamut-kit@0.6.38-alpha.c2129d.0<br/>@codecademy/gamut-labs@28.6.1-alpha.c2129d.0<br/>@codecademy/gamut@42.3.0-alpha.c2129d.0<br/>@codecademy/monaco@0.5.28-alpha.c2129d.0<br/>@codecademy/styleguide@52.5.0-alpha.c2129d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d713cb977a4c508c193e99--gamut-preview.netlify.app>https://61d713cb977a4c508c193e99--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d713cb977a4c508c193e99>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.312283.0<br/>@codecademy/gamut-kit@0.6.38-alpha.312283.0<br/>@codecademy/gamut-labs@28.6.1-alpha.312283.0<br/>@codecademy/gamut@42.3.0-alpha.312283.0<br/>@codecademy/monaco@0.5.28-alpha.312283.0<br/>@codecademy/styleguide@52.5.0-alpha.312283.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d761530562850af0d8452f--gamut-preview.netlify.app>https://61d761530562850af0d8452f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d761530562850af0d8452f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.39-alpha.a7f78e.0<br/>@codecademy/gamut-labs@28.6.2-alpha.a7f78e.0<br/>@codecademy/gamut@42.3.1-alpha.a7f78e.0<br/>@codecademy/monaco@0.5.29-alpha.a7f78e.0<br/>@codecademy/styleguide@52.5.1-alpha.a7f78e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61dc7ec8c9c3001134ec0c2e--gamut-preview.netlify.app>https://61dc7ec8c9c3001134ec0c2e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61dc7ec8c9c3001134ec0c2e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61e07b3a78d1d630aca5867b--gamut-preview.netlify.app>https://61e07b3a78d1d630aca5867b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61e07b3a78d1d630aca5867b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.42-alpha.3f98d6.0<br/>@codecademy/gamut-labs@28.6.5-alpha.3f98d6.0<br/>@codecademy/gamut@43.1.1-alpha.3f98d6.0<br/>@codecademy/monaco@0.5.31-alpha.3f98d6.0<br/>@codecademy/styleguide@53.0.2-alpha.3f98d6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61e831b09290dc3333f64bf5--gamut-preview.netlify.app>https://61e831b09290dc3333f64bf5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61e831b09290dc3333f64bf5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.45-alpha.165c90.0<br/>@codecademy/gamut-labs@28.7.2-alpha.165c90.0<br/>@codecademy/gamut@43.1.3-alpha.165c90.0<br/>@codecademy/monaco@0.5.33-alpha.165c90.0<br/>@codecademy/styleguide@53.1.2-alpha.165c90.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61ef09c5f85e6e7c8a24e26e--gamut-preview.netlify.app>https://61ef09c5f85e6e7c8a24e26e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61ef09c5f85e6e7c8a24e26e>Deploy Logs</a></p>

File: packages/gamut/src/Pagination/index.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+   */
+  navigation?: false;
+  /**
+   * Called when the page number is changed with the resulting page number as its first argument
+   */
+  onChange: (arg0: number) => void;
+  /**
+   *  Basic pagination vs ellipsis variant
+   */
+  type?: 'basic' | 'ellipsis';
```
Comment (jakemhiller) [inline packages/gamut/src/Pagination/index.tsx:37]: idea: add an `automatic` type (maybe the default?) that switches to `ellipses` when there are > X pages?

just a thought, I'm not as familiar w/ the use cases we already have for this.

File: packages/styleguide/stories/Molecules/Pagination/index.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      navigation: true,
+      onChange: (pageNumber) => {
+        window.open(
+          `https://giphy.com/search/hamtaro-page-${pageNumber}`,
+          '_blank'
+        );
+      },
+    }}
+  >
+    {Pagination.bind({})}
```
[the remark this answers is not in the document]
Comment (jakemhiller) [reply packages/styleguide/stories/Molecules/Pagination/index.stories.mdx:113]: Screen Shot 2022-01-27 at 11 03 20 AM

**Review by jakemhiller — APPROVED** [review]
This feels great! Minor issue with the story source, but nothing that should block shipping this version

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fa9bf859935e00b5ee25e2--gamut-preview.netlify.app>https://61fa9bf859935e00b5ee25e2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fa9bf859935e00b5ee25e2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.3e8f17.0<br/>@codecademy/gamut-labs@29.2.5-alpha.3e8f17.0<br/>@codecademy/gamut@44.2.4-alpha.3e8f17.0<br/>@codecademy/monaco@0.6.7-alpha.3e8f17.0<br/>@codecademy/styleguide@54.0.9-alpha.3e8f17.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fac83acf597915f8f477e5--gamut-preview.netlify.app>https://61fac83acf597915f8f477e5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fac83acf597915f8f477e5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.706630.0<br/>@codecademy/gamut-labs@29.2.5-alpha.706630.0<br/>@codecademy/gamut@44.2.4-alpha.706630.0<br/>@codecademy/monaco@0.6.7-alpha.706630.0<br/>@codecademy/styleguide@54.0.9-alpha.706630.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fac909a786e000c29ced81--gamut-preview.netlify.app>https://61fac909a786e000c29ced81--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fac909a786e000c29ced81>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.44aa13.0<br/>@codecademy/gamut-labs@29.2.5-alpha.44aa13.0<br/>@codecademy/gamut@44.2.4-alpha.44aa13.0<br/>@codecademy/monaco@0.6.7-alpha.44aa13.0<br/>@codecademy/styleguide@54.0.9-alpha.44aa13.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fae0d62b0f0a0ae0111e91--gamut-preview.netlify.app>https://61fae0d62b0f0a0ae0111e91--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fae0d62b0f0a0ae0111e91>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.afba61.0<br/>@codecademy/gamut-labs@29.2.5-alpha.afba61.0<br/>@codecademy/gamut@44.2.4-alpha.afba61.0<br/>@codecademy/monaco@0.6.7-alpha.afba61.0<br/>@codecademy/styleguide@54.0.9-alpha.afba61.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61faf2c6e54525007a577a75--gamut-preview.netlify.app>https://61faf2c6e54525007a577a75--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61faf2c6e54525007a577a75>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.72ebf3.0<br/>@codecademy/gamut-labs@29.2.5-alpha.72ebf3.0<br/>@codecademy/gamut@44.2.4-alpha.72ebf3.0<br/>@codecademy/monaco@0.6.7-alpha.72ebf3.0<br/>@codecademy/styleguide@54.0.9-alpha.72ebf3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61faf7adca16090901f6777b--gamut-preview.netlify.app>https://61faf7adca16090901f6777b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61faf7adca16090901f6777b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fafa264df6ad0ced26e978--gamut-preview.netlify.app>https://61fafa264df6ad0ced26e978--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fafa264df6ad0ced26e978>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.54-alpha.3deb14.0<br/>@codecademy/gamut-labs@29.2.5-alpha.3deb14.0<br/>@codecademy/gamut@44.2.4-alpha.3deb14.0<br/>@codecademy/monaco@0.6.7-alpha.3deb14.0<br/>@codecademy/styleguide@54.0.9-alpha.3deb14.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fb0734ef7ab919a62ddec9--gamut-preview.netlify.app>https://61fb0734ef7ab919a62ddec9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fb0734ef7ab919a62ddec9>Deploy Logs</a></p>

**Pull request merged**